kernel/arch/dreamcast/include/dc/g2bus.h

```diff
@@ -31,50 +33,165 @@
 #include <sys/cdefs.h>
 __BEGIN_DECLS
 
+#include <stdint.h>
+#include <arch/irq.h>
 #include <arch/types.h>
 
-/* DMA copy from SH-4 RAM to G2 bus (dir = 0) or the opposite;
-   length must be a multiple of 32,
-   and the source and destination addresses must be aligned on 32-byte
-   boundaries. If block is non-zero, this function won't return until
-   the transfer is complete. If callback is non-NULL, it will be called
-   upon completion (in an interrupt context!). Returns <0 on error.
+#include <dc/fifo.h>
 
-   Known working combination :
+/** \name       List of G2 Bus channels
+ 
+    AICA (SPU) is channel 0, BBA uses channel 1. CH2_DMA_G2CHN and 
+    CH3_DMA_G2CHN are not currently tied to any specific device.
 
-   g2chn = 0, sh4chn = 3 --> mode = 5 (but many other value seems OK ?)
-   g2chn = 1, sh4chn = 1 --> mode = 0 (or 4 better ?)
-   g2chn = 1, sh4chn = 0 --> mode = 3
+    \note 
+      A change in the implementation has rendered *_DMA_MODE and *_DMA_SHCHN 
+      obsolete.
 
-   It seems that g2chn is not important when choosing mode, so this mode parameter is probably
-   how we actually connect the sh4chn to the g2chn.
+      In the current implementation, *_DMA_MODE should always be set to zero 
+      (representing CPU_TRIGGER). There is also no involvement of SH4-DMA with
+      G2-DMA; therefore, the *_DMA_SHCHN values have been deprecated.
+      
+    @{
+*/
+#define SPU_DMA_G2CHN  0 /**< \brief AICA: G2 channel 0 */
+#define SPU_DMA_MODE   0 /**< \brief Unused; Here for compatibility */
+#define SPU_DMA_SHCHN  0 /**< \brief Unused; Here for compatibility */
+
+#define BBA_DMA_G2CHN  1 /**< \brief BBA:  G2 channel 1 */
+#define BBA_DMA_MODE   0 /**< \brief Unused; Here for compatibility */
+#define BBA_DMA_SHCHN  0 /**< \brief Unused; Here for compatibility */
+
+#define CH2_DMA_G2CHN  2 /**< \brief CH2: G2 channel 2 */
+
+#define CH3_DMA_G2CHN  3 /**< \brief CH3: G2 channel 3 */
```

Comment:
Considering the MODE/SHCHN stuff was only ever used internally, we really can get rid of them.

```suggestion
#define SPU_DMA_G2CHN  0 /**< \brief AICA: G2 channel 0 */
#define BBA_DMA_G2CHN  1 /**< \brief BBA:  G2 channel 1 */
#define CH2_DMA_G2CHN  2 /**< \brief CH2: G2 channel 2 */
#define CH3_DMA_G2CHN  3 /**< \brief CH3: G2 channel 3 */
```

Reply:
It might also be nice to rename these... Perhaps `G2_CHAN_SPU_DMA` and the like?

Reply:
Renamed to specifications